### ISSUE TITLE
free conflicts with crtdbg.h free

### ISSUE THREAD

**Issue by Codewyvern** [recastnavigation/recastnavigation#241]
I am including this global header file:

#pragma once

#define _CRTDBG_MAP_ALLOC
#include <stdlib.h>
#include <crtdbg.h>

In order to perform memory leak checking, this creates a free macro which then conflicts with a bunch of methods inside the recast navigation code, quite a lot of them being in DetourTileCacheBuilder.h

I use about 30 different open source libraries in my project and only recast actually triggers this problem.

Would it be possible to rename a few of those methods?

**Comment by hymerman** [comment]
Yes, good idea - do you know which set of functions it is? I guess `free` and `alloc` which are very general names. It's a shame that libraries `#define` these as it basically makes them unusable keywords.

This will have to wait for the next major version since it's technically a breaking interface change. Unless we can `#ifdef` the current functions out in the presence of macro `free` or `alloc` and claim that they were broken in those conditions anyway so it's not a breaking change? Hmm. Still don't fully understand this semantic versioning lark :)

**Comment by Janiels** [comment]
Before including `DetourTileCacheBuilder.h` you can also push the macro and undefine it yourself. I'm not sure this is portable though, but I believe it works on all major compilers:
```
#pragma push_macro("free")
#undef free
#include "DetourTileCacheBuilder.h"
#pragma pop_macro("free")
```
